net/DescriptionViewController.m

```diff
@@ -86,104 +174,527 @@ -(void)issueImageDidLoad
+    
+    self.addCommentDynamicTextView.layer.cornerRadius = self.addCommentDynamicTextView.frame.size.height / 4;
+    
+    [self scrollBottomConstraint].constant = keyboardHeight + textFieldHeight - tabBarHeight;
+    [self.view layoutIfNeeded];
+    
+    [self.view insertSubview:grayView belowSubview:self.addCommentDynamicTextView];
+    [self.view addSubview:addCommentButton];
+    [addCommentButton sizeToFit];
+    
+    [grayView.topAnchor constraintEqualToAnchor:self.ScrollView.bottomAnchor].active = YES;
+    [grayView.leadingAnchor constraintEqualToAnchor:self.view.leadingAnchor].active = YES;
+    [grayView.trailingAnchor constraintEqualToAnchor:self.view.trailingAnchor].active = YES;
+    [grayView.heightAnchor constraintEqualToConstant:textFieldHeight].active = YES;
+    
+    [self.addCommentDynamicTextView.topAnchor constraintEqualToAnchor:grayView.topAnchor constant:2].active = YES;
+    [self.addCommentDynamicTextView.leadingAnchor constraintEqualToAnchor:grayView.leadingAnchor constant:8].active = YES;
+    [self.addCommentDynamicTextView.trailingAnchor constraintEqualToAnchor:addCommentButton.leadingAnchor constant:-8].active = YES;
+    [self.addCommentDynamicTextView.bottomAnchor constraintEqualToAnchor:grayView.bottomAnchor constant:-2].active = YES;
+    
+    [addCommentButton.trailingAnchor constraintEqualToAnchor:grayView.trailingAnchor constant:-8].active = YES;
+    [addCommentButton.centerYAnchor constraintEqualToAnchor:self.addCommentDynamicTextView.centerYAnchor].active = YES;
+    [addCommentButton.widthAnchor constraintEqualToConstant:buttonTittleSize.width+4].active = YES;
+    
+    
+    [UIView animateWithDuration:0.3 animations:^{
+        self.ScrollView.contentOffset = CGPointMake(0, self.contentView.frame.size.height - self.ScrollView.frame.size.height);
+    }];
+    
+    
+    
+    
+}
+
+-(void)keyboardWillHide
+{
+    [self scrollBottomConstraint].constant = 0;
+    [self.view layoutIfNeeded];
+    
+}
+
+-(void)sendCommentPressed
+{
+    NSString *message = self.addCommentDynamicTextView.text;
+    [self.addCommentDynamicTextView resignFirstResponder];
+    [self.addCommentDynamicButton removeFromSuperview];
+    [self.addCommentDynamicGreyView removeFromSuperview];
+    [self.addCommentDynamicGreyView removeFromSuperview];
+    
+    self.addCommentDynamicButton = nil;
+    self.addCommentDynamicButton = nil;
+    self.addCommentDynamicButton = nil;
+    
+    [self requestUsersAndAddNewCommentsAndSendMessage:message];
+    
+}
+
+-(void)requestUsersAndAddNewCommentsAndSendMessage:(NSString*)message
+{
+    __weak DescriptionViewController *weakSelf = self;
+    [self.dataSorce requestAllUsers:^(NSArray<NSDictionary<NSString *,NSString *> *> *userDictionaries) {
+        [weakSelf sendMessage:message andAddnewCommentsBlockWithAllUserDictionaries:userDictionaries];
+    } withErrorHandler:^(NSError *error) {
```

Comment:
Excelent question. I've chosen wrong way with implementing model methods. Its make sence to pass either NSArray<NSDictionary<NSString *,NSString *> *> *userDictionaries and NSError *error in one block. I'll refactor it.
About NSError: for out project its enough to check recieved in block data, and if its nill - than somethig is wrong.